### ISSUE TITLE
Implement document limits 

### ISSUE THREAD

**Issue by amorton** [stargate/jsonapi#173]
Following from spec work in #135 

Any document, on insert or after being updated, should be checked again the document rules. 

Document limits are under the [Documents](https://github.com/stargate/jsonapi/blob/main/docs/jsonapi-spec.textile#documents) section in the spec and include: 

- field name limits 
- and limits on the complexity and size of documents in [Document Limits](https://github.com/stargate/jsonapi/blob/main/docs/jsonapi-spec.textile#document-limits)

If a document breaks any rule we should not store it. 

If this is happening in an updateMany operation we should not store any of the updated documents. 

Work may be blocked by deciding on rules for numbers in #172

**Comment by amorton** [comment]
@tatu-at-datastax  assigning to you as I assume this is in shredding / de-shredding

**Comment by tatu-at-datastax** [comment]
@amorton Yes makes sense was expecting to work on this. Will be interesting combination of validating at different stages, and happens to also align (surprisingly!) with some Jackson OSS work wrt limits -- which are mostly new additions in upcoming Jackson 2.15 (this is sort of hot topic right now; similar limits have existed in XML for a while, now coming to JSON processing).
